File: blotztask-ui/src/app/dashboard/task-list/models/scheduled-tasks-dto.ts
import { TaskDetailDTO } from "./task-detail-dto";

export interface ScheduledTasksDTO {
  overdueTasks: TaskDetailDTO[];
  todayTasks: TaskDetailDTO[]; 
  tomorrowTasks: TaskDetailDTO[]; 
  weekTasks: TaskDetailDTO[]; 
  monthTasks: Record<number, TaskDetailDTO[]>; 
}

File: blotztask-api/Modules/Tasks/DTOs/ScheduledTasksDTO.cs
namespace BlotzTask.Modules.Tasks.DTOs;

public class ScheduledTasksDto
{
        
    public List<TaskItemDto> OverdueTasks { get; set; }
    public List<TaskItemDto> TodayTasks { get; set; }
    public List<TaskItemDto> TomorrowTasks { get; set; }
    public List<TaskItemDto> WeekTasks { get; set; }
    public List<TaskItemDto> FloatingTasks { get; set; }

    public Dictionary<int, List<TaskItemDto>> MonthTasks { get; set; }
}